lib/active_record_host_pool/connection_adapter_mixin.rb

```diff
@@ -87,9 +87,17 @@ module ConnectionAdapters
     class ConnectionHandler
 
       if ActiveRecord::VERSION::MAJOR >= 5
```

Comment:
maybe change to == 5 to be safe ... otherwise 6.0 will trigger you new code path